### PR TITLE
Feature/354 Add Stream as content

### PR DESCRIPTION
**Pull request by Kingmidas74**

### CODE REVIEW

**Comment by Kingmidas74** [comment]
I've added Stream as option to Content and now not all tests are passed.

File: src/NClient.Providers/NClient.Providers.Transport.SystemNetHttp/SystemNetHttpTransportRequestBuilder.cs [hunk 1]
```diff
@@ -42,7 +42,7 @@ public Task<HttpRequestMessage> BuildAsync(IRequest request, CancellationToken c
 
             if (request.Content is not null)
             {
-                httpRequestMessage.Content = new ByteArrayContent(request.Content.Bytes);
+                httpRequestMessage.Content = new StreamContent(request.Content.GetContentAsStream());
```
Comment (smolchanovsky) [inline src/NClient.Providers/NClient.Providers.Transport.SystemNetHttp/SystemNetHttpTransportRequestBuilder.cs:45]: Why wrap in `StreamContent`?

File: src/NClient/NClient.Abstractions/Providers/Transport/Common/Content.cs [hunk 2]
```diff
@@ -26,9 +35,13 @@ public class Content : IContent
         public IMetadataContainer Metadatas { get; }
 
         [SuppressMessage("ReSharper", "UnusedVariable")]
-        public Content(byte[]? bytes = null, string? encoding = null, IMetadataContainer? headerContainer = null)
+        public Content(byte[]? bytes = null, Stream? streamContent = null, string? encoding = null, IMetadataContainer? headerContainer = null)
```
Comment (smolchanovsky) [inline src/NClient/NClient.Abstractions/Providers/Transport/Common/Content.cs:38]: Why do we need bytes if there is a stream now?

**Review by smolchanovsky — CHANGES_REQUESTED** [review]

**Comment by smolchanovsky** [comment]
> I've added Stream as option to Content and now not all tests are passed.

Why don't they pass? What error?

**Review by smolchanovsky — APPROVED** [review]

**Review by smolchanovsky — APPROVED** [review]

**Pull request merged**